src/domain/model/authentication/AssignmentActions.ts

```diff
@@ -0,0 +1,93 @@
+import { AssignmentAction, AssignmentId } from './AssignmentAction';
+import { AssignmentActionType } from './AssignmentActionType';
+import { UserId } from '../user/UserId';
+
+export class AssignmentActions<T, R> {
```

Comment:
My main question with this is whether we need the generics, or whether we could achieve the same thing with a base class?   It's very hard to follow to follow at the moment.

I would even consider a comment above the class definition.
```
// R represents the Resource, T represents what it's assigned To
export class AssignmentActions<T, R> {
```
Otherwise I need to read quite far into the class to work out what these generics represent.

Reply:
I realise there's a convention to use single letters, but I don't believe that's a constraint.  Would you consider naming them
`export class AssignmentActions<To, Resource> {`
?

I think that would be more readable rather than less.

Reply:
Yeah, I usually try to stick to the Generics naming conventions but I agree that in this case it might make more sense to follow class name conventions.